tensorboard/webapp/widgets/card_fob/card_fob_types.ts

```diff
@@ -16,8 +16,8 @@ limitations under the License.
  * The start and end time to define a time selection, used for linked time and step selector.
  */
 export interface TimeSelection {
-  start: {step: number};
-  end: {step: number} | null;
+  start: {step: number} | null;
```

Comment:
The decision to have end be `end: {step: number} | null` rather than `end?: {step: number}` has lead to a lot of clunky code (especially in tests). I would love to see this changed and I think this may be a good time for this update. However, I recognize that this is not a small change so I'll leave it to your discression.